OpenRA.Mods.Common/WorldExtensions.cs

```diff
@@ -11,12 +11,191 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using OpenRA.Mods.Common.Traits;
 
 namespace OpenRA.Mods.Common
 {
 	public static class WorldExtensions
 	{
+		/// <summary>
+		/// Filters <paramref name="actors"/> by only returning those that can be reached as the target of a path from
+		/// <paramref name="sourceActor"/>. Only terrain is taken into account, i.e. as if
+		/// <see cref="BlockedByActor.None"/> was given.
+		/// <paramref name="targetOffsets"/> is used to define locations around each actor in <paramref name="actors"/>
+		/// of which one must be reachable.
+		/// </summary>
+		public static IEnumerable<(Actor Actor, WVec[] ReachableOffsets)> WithPathFrom(this IEnumerable<Actor> actors, Actor sourceActor, Func<Actor, WVec[]> targetOffsets)
+		{
+			if (sourceActor.TraitOrDefault<Aircraft>() != null)
+				return actors.Select<Actor, (Actor Actor, WVec[] ReachableOffsets)>(a => (a, targetOffsets(a)));
+			var mobile = sourceActor.TraitOrDefault<Mobile>();
```

Comment:
```suggestion
			if (sourceActor.Info.HasTraitInfo<AircraftInfo>())
				return actors.Select<Actor, (Actor Actor, WVec[] ReachableOffsets)>(a => (a, targetOffsets(a)));
			
			var mobile = sourceActor.TraitOrDefault<Mobile>();
```

If we don't use the variable it's better to retrieve the boolean directly

Reply:
Updated all.